### DISCUSSION TITLE
seccomp user-space notification addfd

### DISCUSSION THREAD

**Discussion by rusty-snake** [libseccomp-rs/libseccomp-rs#153] [Show and tell]
While libseccomp does not provide a wrapper for `SECCOMP_IOCTL_NOTIF_ADDFD` yet (https://github.com/seccomp/libseccomp/issues/380), you can actually use it. If anyone wants to hack with it, fell free to copy over the modules.

- [libseccomp_sys_ext_unotify_addfd.rs](https://github.com/rusty-snake/openat_dialog_poc/blob/bfd89ca6a60a21e55f61c2e677768cdc03e3db9c/src/libseccomp_sys_ext_unotify_addfd.rs)
- [libseccomp_ext_unotify_addfd.rs](https://github.com/rusty-snake/openat_dialog_poc/blob/bfd89ca6a60a21e55f61c2e677768cdc03e3db9c/src/libseccomp_ext_unotify_addfd.rs)

Note, `ioctl` is ~a bit~ more than a bit of a mess so IDK if the value of `SECCOMP_IOCTL_NOTIF_ADDFD` is architecture depended or not.

**Comment by ManaSugi** [comment]
This is very nice for users (including me) who want to use the addfd features! Thank you!
